plugins/woocommerce/src/Admin/API/LaunchYourStore.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * REST API Launch Your Store Controller
+ *
+ * Handles requests to /launch-your-store/*
+ */
+
+namespace Automattic\WooCommerce\Admin\API;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Launch Your Store controller.
+ *
+ * @internal
+ */
+class LaunchYourStore {
+
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wc-admin';
+
+	/**
+	 * Route base.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'launch-your-store';
+
+	/**
+	 * Register routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/dismiss-coming-soon-banner',
```

Comment:
What do you think about making this more generic so it could be re-used for other dismissible items too?

Perhaps something like `'/' . $this->rest_base . '/dismiss-message'` and allow `coming-soon-banner` to be passed as a parameter.

Reply:
I think..we're creating another version https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Admin/API/Options.php, which is deprecated now.

In my opinion, it's safer (since we define permission separately) and easier to understand with a dedicate endpoint. I agree that this might add more codes though 🥲 

Do you have other use-cases that require a generic endpoint?

Reply:
I think it's a bit premature to implement a generic dismiss API in this PR. Main reason is I don't think it's established that we only set an option to `no`, there could be other things to do for other use cases.

If we ever do have one, I think the better way is analyse similar use cases in admin and then refactor all of them instead.

Reply:
@moon0326  @ilyasfoo  Good points, thank you!

Come to think of it -- now that we're using user meta, would we be able to just use the in-built [user endpoint](https://developer.wordpress.org/rest-api/reference/users/)?

> Do you have other use-cases that require a generic endpoint?

I was thinking it could be used elsewhere on any items that need to be dismissed. The [LYS tour](https://github.com/woocommerce/woocommerce/pull/46203) could be another use-case. But let's shelve this idea.

Reply:
@adrianduffell That's a good point! Nice!

Yup! We can use the in-built endpoint 👍

Reply:
@adrianduffell updated in [Use the built-in rest api to manage woocommerce_coming_soon_banner_di…](https://github.com/woocommerce/woocommerce/pull/46149/commits/e0edbe0d823462e2f9486aef18f183df2293d6a6)